File: test_api.py
import requests
import json

headers = {
    "Site-Id": "5",
    "Referer": "https://v5.animelib.org/",
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) Chrome/134.0.0.0",
    "Accept": "application/json"
}

resp = requests.get("https://api.cdnlibs.org/api/anime?site_id[]=5&page=1", headers=headers)
with open("response.json", "w", encoding="utf-8") as f:
    json.dump(resp.json(), f, ensure_ascii=False, indent=2)

File: test_api.rs
use reqwest::blocking::Client;
use reqwest::header;

fn main() {
    let mut headers = header::HeaderMap::new();
    headers.insert("Site-Id", header::HeaderValue::from_static("5"));
    headers.insert(header::REFERER, header::HeaderValue::from_static("https://v5.animelib.org/"));
    headers.insert(header::USER_AGENT, header::HeaderValue::from_static("Mozilla/5.0 (Windows NT 10.0; Win64; x64) Chrome/134.0.0.0"));
    headers.insert(header::ACCEPT, header::HeaderValue::from_static("application/json"));
    
    let client = Client::builder().default_headers(headers).build().unwrap();
    let resp = client.get("https://api.cdnlibs.org/api/anime?site_id[]=5&page=1").send().unwrap();
    let text = resp.text().unwrap();
    
    std::fs::write("response.json", &text).unwrap();
    println!("Saved to response.json");
}
